rpc/src/v1/impls/personal.rs

```diff
@@ -106,28 +95,41 @@ impl<C, M> Personal for PersonalClient<C, M> where
 	}
 
 	fn send_transaction(&self, meta: Metadata, request: TransactionRequest, password: String) -> BoxFuture<RpcH256, Error> {
-		let sign_and_send = move || {
-			let client = take_weak!(self.client);
-			let miner = take_weak!(self.miner);
+		let setup = || {
+			let dispatcher = self.dispatcher.clone();
 			let accounts = take_weak!(self.accounts);
```

Comment:
Perhaps it's worth to introduce `take_weak` alike macro for futures to avoid such `setup` steps.